addons/sale_timesheet/models/account_move_reversal.py

```diff
@@ -0,0 +1,14 @@
+from odoo import models
+
+class AccountMoveReversal(models.TransientModel):
+    _inherit = 'account.move.reversal'
+
+    def reverse_moves(self):
+        result = super(AccountMoveReversal, self).reverse_moves()
+        moves = self.move_ids
+        timesheet_lines = self.env['account.analytic.line'].search([
+            ('timesheet_invoice_id', 'in', moves.ids)
+        ])
+        if timesheet_lines:
+            timesheet_lines.write({'timesheet_invoice_id': False})
+        return result
```

Comment:
Cannot we remove the invoice set on the timesheets, only when the credit note for that invoice is confirmed?

By doing that, we could check which products are inside the credit note to know if we should or not remove the timesheets.

Here I have the feelings we will always remove the timesheets even if we create a credit note to partially reserve the invoice, right?